modules/plotting/src/interaction/boxselectioninteractionhandler.cpp

```diff
@@ -0,0 +1,205 @@
+        [min = start[0], max = end[0]](auto brprecision) {
+            using ValueType = util::PrecisionValueType<decltype(brprecision)>;
+            std::vector<size_t> selectedIndices;
+            selectedIndices.reserve(brprecision->getSize());
+            for (auto&& [ind, elem] : util::enumerate(brprecision->getDataContainer())) {
+                if (static_cast<double>(elem) < min || static_cast<double>(elem) > max) {
+                    continue;
+                } else {
+                    selectedIndices.emplace_back(ind);
+                }
+            }
+            return selectedIndices;
+        });
+    // Use indices filted by x-axis as input
+    auto ybuf = yAxis->getRepresentation<BufferRAM>();
+    auto selectedIndices = ybuf->dispatch<std::unordered_set<size_t>, dispatching::filter::Scalars>(
+        [selectedIndicesX, min = start[1], max = end[1]](auto brprecision) {
+            using ValueType = util::PrecisionValueType<decltype(brprecision)>;
+            auto data = brprecision->getDataContainer();
+            std::unordered_set<size_t> selectedIndices;
+            selectedIndices.reserve(selectedIndicesX.size());
+
+            std::copy_if(selectedIndicesX.begin(), selectedIndicesX.end(),
+                         std::inserter(selectedIndices, selectedIndices.begin()),
+                         [data, min, max](auto idx) {
+                             return !(static_cast<double>(data[idx]) < min ||
+                                      static_cast<double>(data[idx]) > max);
+                         });
+
+            return selectedIndices;
+        });
+    return selectedIndices;
+}
+
+std::unordered_set<size_t> BoxSelectionInteractionHandler::boxFilter(const dvec2& start,
+                                                                     const dvec2& end,
+                                                                     const BufferBase* xAxis,
+                                                                     const BufferBase* yAxis) {
+    if (xAxis == nullptr || yAxis == nullptr) {
+        return std::unordered_set<size_t>();
+    }
+    auto xbuf = xAxis->getRepresentation<BufferRAM>();
+    std::unordered_set<size_t> filteredIndices =
+        xbuf->dispatch<std::unordered_set<size_t>, dispatching::filter::Scalars>(
+            [start, end, ybuf = yAxis->getRepresentation<BufferRAM>()](auto brprecision) {
+                using ValueType = util::PrecisionValueType<decltype(brprecision)>;
+                return ybuf->dispatch<std::unordered_set<size_t>, dispatching::filter::Scalars>(
+                    [start, end, xData = brprecision->getDataContainer()](auto brprecision) {
+                        std::unordered_set<size_t> selectedIndices;
+                        for (auto&& [ind, xVal, yVal] :
+                             util::enumerate(xData, brprecision->getDataContainer())) {
+                            if (static_cast<double>(xVal) < start[0] ||
```

Comment:
you might what to cast start and end to T outsize of the loop instead... converting to double is quite expensive.

Reply:
I did so first but then I figured it is not safe because the start/end will be rounded off, which will not produce the same result as casting the value.

Reply:
should be possible, but one has to do the rounding a bit more carefully...

Reply:
Suggestions on how are welcome :)
Simply std::round might work?

Reply:
Or round up for start and down for end?

Reply:
Yes but only in the case that T is not floating point...

Reply:
Fixed